### DISCUSSION TITLE
How to assign values ​​to block tags in DWG files

### DISCUSSION THREAD

**Discussion by fly-sure** [LibreDWG/libredwg#1022] [Q&A]
The picture shows the blocks created in the dwg file. I want to assign values ​​to the tags in the blocks. How can I do this?
![eed06e9092b8ec876437fed98f6d612](https://github.com/user-attachments/assets/43c22656-5128-4e11-91ab-01c9a6e36019)
![微信图片_20240924144219](https://github.com/user-attachments/assets/5e111163-2fa5-4252-a406-efee5c765446)

**Comment by rurban** [comment]
Similar to TEXT, but these are ATTRIB's next to the INSERT

**Reply by fly-sure** [reply]
I have found these contents in the **text_value** of the **Dwg_Entity_ATTRIB** structure. Your suggestions are very useful to me. Thank you very much

**Reply by fly-sure** [reply]
I want to get the complete information, but I only get the first character, and Chinese cannot be output correctly. Is there any solution?
![f19ff5a9d0bb350092e72b67cca8c4b](https://github.com/user-attachments/assets/7c0a3ffb-2191-4b81-b374-a429106c5bb6)
![317bd154cd4475eb610c0c5b9cc252a](https://github.com/user-attachments/assets/9e2f4228-270d-4797-8c55-9a256abfc654)
![5244da3e691478fcfe389fd68ec2dcd](https://github.com/user-attachments/assets/072eca6f-6ff5-480f-a323-efdcef5f7391)

**Reply by fly-sure** [reply]
I found that the **dwg_write_file** function in the example is used to write a new dwg file. If I want to write the value of text_value in the original file, what function should I call?
![Uploading 053bebcf4312d06c7e9ccdc621f3e45.png…]()
